src/include/optimizer/optimizer_metadata.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+
+#include <utility>
+#include <vector>
+
+#include "common/settings.h"
+#include "optimizer/cost_model/abstract_cost_model.h"
+#include "optimizer/group_expression.h"
+#include "optimizer/memo.h"
+#include "optimizer/rule.h"
+#include "optimizer/statistics/stats_storage.h"
+
+namespace terrier {
+
+namespace transaction {
+class TransactionContext;
+}
+
+namespace catalog {
+class CatalogAccessor;
+}
+
+namespace optimizer {
+
+class OptimizerTaskPool;
+class RuleSet;
+
+/**
+ * OptimizerMetadata is a class containing pointers to various objects
+ * that are required during the entire query optimization process.
+ */
+class OptimizerMetadata {
```

Comment:
Should we rename this to `OptimizerContext` to match the other objects in the system?